src/riak_kv_compile_tab.erl

```diff
@@ -112,11 +113,24 @@ insert(BucketType, DDL) when is_binary(BucketType), is_tuple(DDL) ->
 
 %% insert into the v2 table so that the record is available
 insert_v2(BucketType, #ddl_v1{} = DDL) ->
-    %% the version is always 1 for the v2 table
-    DDLVersion = 1,
     %% put compiling as the compile state in the old table so that
     %% it will always recompile the modules on a downgrade
     CompileState = compiling,
+    insert_previous(BucketType, DDL, CompileState).
+
+convert_ddl_v2_to_ddl_v1(#ddl_v2{} = DDL) ->
+    #ddl_v1{
+       table = DDL#ddl_v2.table,
+       fields = DDL#ddl_v2.fields,
+       partition_key = DDL#ddl_v2.partition_key,
+       local_key = DDL#ddl_v2.local_key
+      }.
```

Comment:
Not all tables can be downgraded and be used by TS 1.4, for example tables using the BLOB type or descending keys. If the DDL conversion is used from riak_ql_ddl then this is checked as part of the conversion. If the table cannot be downgraded then it will not be usable on this node. The `riak_ql_ddl_rec_version` capability stop it being used on 1.5 nodes too.